### PR TITLE
Create LSAT on button click

### PR DESCRIPTION
**Pull request by mnicpt**
### Description
Fixes cached lsat for customers receiving 403 errors on capture.

### CODE REVIEW

File: src/api/auth.js [hunk 1]
```diff
@@ -24,6 +25,9 @@ export function createAccessToken(clientID : string, { targetSubject } : Generat
         const data : Object = {
             grant_type: `client_credentials`
         };
+        if (!cache) {
```
Comment (TNuzzi) [inline src/api/auth.js:28]: Not a functional comment just laughed at, if no then yes.

File: src/button/pay.js [hunk 2]
```diff
@@ -107,8 +107,17 @@ export function initiatePaymentFlow({ payment, serviceData, config, components,
...
+        
+        const accessTokenPromise = ZalgoPromise.try(() => {
+            return createAccessToken(clientID, { cache: false })
+                .then(facilitatorAccessToken => facilitatorAccessToken);
+        });
 
-        const clickPromise = click ? ZalgoPromise.try(click) : ZalgoPromise.resolve();
+        const clickPromise = click
+            ? ZalgoPromise.try(() => {
+                return accessTokenPromise.then(click);
```
Comment (TNuzzi) [inline src/button/pay.js:118]: I don't think it is a big deal or even an issue but `click` could return `false` and in that case we would not need to create LSAT token.  I am not sure how much we want to optimize it.

**Comment by TNuzzi** [comment]
Other than my comment this looks good.

**Review by TNuzzi — APPROVED** [review]

File: src/button/pay.js [hunk 3]
```diff
@@ -107,8 +107,17 @@ export function initiatePaymentFlow({ payment, serviceData, config, components,
                 [FPTI_KEY.IS_VAULT]:        instrumentType ? '1' : '0',
                 [FPTI_CUSTOM_KEY.INFO_MSG]: enableNativeCheckout ? 'tester' : ''
             }).flush();
+        
+        const accessTokenPromise = ZalgoPromise.try(() => {
```
Comment (bluepnume) [inline src/button/pay.js:111]: Little confused, we're creating this token but then not passing it anywhere. So this is a noop right now right?

Comment (mnicpt) [reply src/button/pay.js:111]: @bluepnume It's in the `clickPromise`.

Comment (mnicpt) [reply src/button/pay.js:111]: I refactored this and saw that I didn't need it at all.

File: src/button/pay.js (hunk 2 above)
Comment (mnicpt) [reply src/button/pay.js:118]: No good point!

**Pull request closed without merging**